### PR TITLE
Adds some suicide acts

### PR DESCRIPTION
**Pull request by swindly**
:cl: Swindly
add: You can kill yourself with a few more items.
/:cl:

[why]: (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

suicide should always be encouraged

Adds suicide_act() for:

- glimmering residue
- cheap plastic imitation of the nuclear authentication disk
- body bag
- bag of dice
- die
- bruise pack
- pill bottle
- plutonium core
- pack of watermelon seeds

### CODE REVIEW

**Comment by Onule** [comment]
Wait, you can pick up revenant remains, aka glimmering residue?

**Comment by swindly** [comment]
You can pick up revenant remains. That's how you prevent revenants from respawning.

File: code/game/objects/items/storage/firstaid.dm [hunk 1]
```diff
@@ -167,6 +167,10 @@
 				usr.s_active.close(usr)
 			src.show_to(usr)
 
+/obj/item/storage/pill_bottle/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is trying to get the cap off [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
Comment (Jordie0608) [inline code/game/objects/items/storage/firstaid.dm]: 1. Remove cap from pill bottle
2. Die
3. ???
4. ~~Profit!~~ Still death

This one is kinda lame; it'd also logically mean anytime you open a pill bottle you should die.

Comment (swindly) [reply code/game/objects/items/storage/firstaid.dm]: It's a reference to a movie that I thought most people who browsed /tg/ and the rest of 9gag had seen.

Comment (Jordie0608) [reply code/game/objects/items/storage/firstaid.dm]: >In the end, Taylor, Samantha, Cheyenne, Caleb, and Scott all stand up against Lindsay. Other classmates witnessing Taylor standing up to Lindsay realize how bullying is terrible, and antagonize Lindsay for her abuse. A law is then passed against cyberbullying.

Was this really made

File: code/game/gamemodes/nuclear/nuclearbomb.dm [hunk 2]
```diff
@@ -561,3 +561,14 @@ This is here to make the tiles around the station mininuke change when it's arme
...
+/obj/item/disk/fakenucleardisk/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is pretending to go delta! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	playsound(user.loc, 'sound/machines/alarm.ogg', 50, -1, 1)
+	for(var/i in 1 to 100)
+		addtimer(CALLBACK(user, /atom/proc/add_atom_colour, (i % 2)? "#00FF00" : "#FF0000", ADMIN_COLOUR_PRIORITY), i)
+	addtimer(CALLBACK(user, /atom/proc/remove_atom_colour, ADMIN_COLOUR_PRIORITY), 101)
+	addtimer(CALLBACK(user, /atom/proc/visible_message, "<span class='suicide'>[user] was destroyed by the nuclear blast!</span>"), 101)
+	addtimer(CALLBACK(user, /mob/living/proc/adjustOxyLoss, 200), 101)
+	addtimer(CALLBACK(user, /mob/proc/death, 0), 101)
+	return MANUAL_SUICIDE
```
Comment (coiax) [inline code/game/gamemodes/nuclear/nuclearbomb.dm]: I appreciate the use of my nuke disk suicide code, but this is copy pasta...

Maybe we can make nuke suicide a component, honk

File: code/modules/hydroponics/grown/melon.dm [hunk 3]
```diff
@@ -14,6 +14,13 @@
 	mutatelist = list(/obj/item/seeds/watermelon/holy)
 	reagents_add = list("water" = 0.2, "vitamin" = 0.04, "nutriment" = 0.2)
 
+/obj/item/seeds/watermelon/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is swallowing [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	user.gib()
+	new product(loc)
+	qdel(src)
+	return (BRUTELOSS)
```
Comment (coiax) [inline code/modules/hydroponics/grown/melon.dm]: Shouldn't this be MANUAL_SUICIDE

**Comment by CPTANT** [comment]
Hmmm PR's like this do make me wonder why every suicide act is defined at the object itself, instead of all being in a single file.

File: code/game/gamemodes/miniantags/revenant/revenant.dm [hunk 4]
```diff
@@ -411,6 +411,10 @@
 	revenant = null
 	qdel(src)
 
+/obj/item/ectoplasm/revenant/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is inhaling [src]! It looks like [user.p_theyre()] trying to visit the shadow realm!</span>")
```
Comment (Alek2ander) [inline code/game/gamemodes/miniantags/revenant/revenant.dm]: Should scatter the ectoplasm, probably.

File: code/game/objects/items/storage/firstaid.dm (hunk 1 above)
Comment (Buggy123) [reply code/game/objects/items/storage/firstaid.dm]: Holy shit all it's missing is a "And then everyone in the classroom clapped."

File: code/game/gamemodes/nuclear/nuclearbomb.dm (hunk 2 above)
Comment (Jalleo) [reply code/game/gamemodes/nuclear/nuclearbomb.dm]: Would be a suitable solution.

File: code/modules/hydroponics/grown/melon.dm [hunk 5]
```diff
@@ -14,6 +14,13 @@
 	mutatelist = list(/obj/item/seeds/watermelon/holy)
 	reagents_add = list("water" = 0.2, "vitamin" = 0.04, "nutriment" = 0.2)
 
+/obj/item/seeds/watermelon/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is swallowing [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	user.gib()
```
Comment (Jalleo) [inline code/modules/hydroponics/grown/melon.dm]: gibbing is probably a bad idea.

Comment (swindly) [reply code/modules/hydroponics/grown/melon.dm]: C4 also gibs on suicide

File: code/game/gamemodes/nuclear/nuclearbomb.dm [hunk 6]
```diff
@@ -561,3 +561,11 @@ This is here to make the tiles around the station mininuke change when it's arme
 	name = "cheap plastic imitation of the nuclear authentication disk"
 	desc = "Broken dreams and a faint odor of cheese."
 	icon_state = "nucleardisk"
+
+/obj/item/disk/fakenucleardisk/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is pretending to go delta! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	playsound(user.loc, 'sound/machines/alarm.ogg', 30, -1, 1)
```
Comment (Cyberboss) [inline code/game/gamemodes/nuclear/nuclearbomb.dm]: playsound(src, ...

File: code/game/gamemodes/nuclear/nuclearbomb.dm [hunk 7]
```diff
@@ -561,3 +561,11 @@ This is here to make the tiles around the station mininuke change when it's arme
...
+	addtimer(CALLBACK(user, /atom/proc/visible_message, "<span class='suicide'>[user] was destroyed by the nuclear blast!</span>"), 101)
+	addtimer(CALLBACK(user, /mob/living/proc/adjustOxyLoss, 200), 101)
+	addtimer(CALLBACK(user, /mob/proc/death, 0), 101)
```
Comment (Cyberboss) [inline code/game/gamemodes/nuclear/nuclearbomb.dm]: These should be one timer with a proc that calls all 3 functions

File: code/game/objects/items/bodybag.dm [hunk 8]
```diff
@@ -21,6 +21,16 @@
...
 	qdel(src)
 
+/obj/item/bodybag/suicide_act(mob/user)
+	if(isopenturf(user.loc))
+		user.visible_message("<span class='suicide'>[user] is crawling into [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		var/obj/structure/closet/body_bag/R = new unfoldedbag_path(user.loc)
+		R.add_fingerprint(user)
+		qdel(src)
+		user.forceMove(R)
+		playsound(loc, 'sound/items/zip.ogg', 15, 1, -3)
```
Comment (Cyberboss) [inline code/game/objects/items/bodybag.dm]: loc -> src

File: code/modules/hydroponics/grown/melon.dm [hunk 9]
```diff
@@ -14,6 +14,13 @@
...
+	new product(loc)
```
Comment (Cyberboss) [inline code/modules/hydroponics/grown/melon.dm]: loc -> drop_location()

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/hydroponics/grown/melon.dm (hunk 5 above)
Comment (octareenroon91) [reply code/modules/hydroponics/grown/melon.dm]: c4 is a bit rarer though

Comment (MrDoomBringer) [reply code/modules/hydroponics/grown/melon.dm]: oxygen tanks gib at max pressure, and they arent rare. how is this any worse?

File: code/game/gamemodes/nuclear/nuclearbomb.dm [hunk 10]
```diff
@@ -548,14 +548,17 @@ This is here to make the tiles around the station mininuke change when it's arme
...
 	for(var/i in 1 to 100)
 		addtimer(CALLBACK(user, /atom/proc/add_atom_colour, (i % 2)? "#00FF00" : "#FF0000", ADMIN_COLOUR_PRIORITY), i)
-	addtimer(CALLBACK(user, /atom/proc/remove_atom_colour, ADMIN_COLOUR_PRIORITY), 101)
-	addtimer(CALLBACK(user, /atom/proc/visible_message, "<span class='suicide'>[user] was destroyed by the nuclear blast!</span>"), 101)
-	addtimer(CALLBACK(user, /mob/living/proc/adjustOxyLoss, 200), 101)
-	addtimer(CALLBACK(user, /mob/proc/death, 0), 101)
+	addtimer(CALLBACK(src, .proc/manual_suicide, user), 101)
 	return MANUAL_SUICIDE
+	
+/obj/item/disk/nuclear/proc/manual_suicide(mob/living/user)
```
Comment (Cyberboss) [inline code/game/gamemodes/nuclear/nuclearbomb.dm]: Should be on /disk to remove duplicated code

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**